node/flatpak-node-generator.py

```diff
@@ -1405,14 +1385,12 @@ async def generate_package(self, package: Package) -> None:
 
         elif isinstance(source, GitSource):
             repo_name = urllib.parse.urlparse(source.url).path.split('/')[-1]
-            tarball_url = source.tarball_url
-            assert tarball_url is not None
-            metadata = await RemoteUrlMetadata.get(tarball_url, cachable=True)
-            filename = f'{repo_name}-{source.commit}'
+            repo_dir = self.gen.tmp_root / repo_name
+            target_tar = self.mirror_dir / f'{repo_name}-{source.commit}'
 
             #XXX Yarn wants uncompressed tar, why?
-            self.gen.add_url_source(tarball_url, metadata.integrity, self.mirror_dir / f'{filename}.gz')
-            self.gen.add_command(f'gunzip "{self.mirror_dir}/{filename}.gz"')
+            self.gen.add_git_source(source.url, source.commit, repo_dir)
+            self.gen.add_command(f'cd {repo_dir}; git archive --format tar -o ../../../{target_tar} HEAD')
```

Comment:
...or create tarball from specific commit here instead of HEAD.

Reply:
The source is already at the commit, so HEAD points to the desired commit.

Reply:
It won't necessarily be at correct commit if there are multiple instances of the same module with different commits.

Reply:
Yeah but this will be fixed if we do what you said in your review above.